src/app/static/topic-experts/topic-experts.component.ts

```diff
@@ -0,0 +1,41 @@
+import { Component, OnInit } from '@angular/core';
+import { User } from 'src/app/models/User';
+import { UsersService } from 'src/app/services/users.service';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'app-topic-experts',
+  templateUrl: './topic-experts.component.html',
+  styleUrls: ['./topic-experts.component.css']
+})
+
+export class TopicExpertsComponent implements OnInit {
+  experts: User[];
+  expert: User = null;
+  returnExpert: User = null;
+  returnNames: User[] = [];
+  getTopicId: number = 0;
+  constructor(private usersService: UsersService,private route: ActivatedRoute) {}
+
+  ngOnInit() {
+    if (this.route.snapshot.paramMap.get('id') !== null) {
+      this.getTopicId = +this.route.snapshot.paramMap.get('id');
+    }
+    this.usersService.getExpertTags().subscribe(expert => {
+      this.experts = expert;
+      this.filterExperts();
+    });
+  }
+
+  filterExperts(): void {
+    for (let i = 0; i < this.experts.length; i++) {
+      this.expert = this.experts[i];
+      for (let j = 0; j < this.expert.expertTags.length; j++) {
+        if (this.expert.expertTags[j].id === this.getTopicId) {
```

Comment:
This logic should be cleaned up.  Don't use fields when local variables will work, and make sure the variable names actually reflect what values are stored in them.

Reply:
done